### ISSUE TITLE
MyList can show too much

### ISSUE THREAD

**Issue by rjledger** [Saleslogix/argos-saleslogix#149]
MyList (of Activities.. and hopefully events - soon as well ;-) can be a bit cumbersome to use if there are a reasonable number of "future" activities on the list. You have to either scroll or manually collapse the (section) headers. It would really improve the user experience and usefulness of this datetime grouping if we had the option to set a "collapse true" on a section-by-section grouping.

**Comment by jbest84** [comment]
@rjledger Cameron has a pull request that will change this. It will insert a new #today hash tag as the default when the list loads. You can open the hash tags to select tomorrow, yesterday, or perhaps even clear the search to show all. Here is a link to the changeset: https://github.com/Saleslogix/argos-saleslogix/pull/148/files

**Comment by rjledger** [comment]
@Jason – I saw this and the idea of a default hashtag (for ALL entity lists) would be a good user productivity improvement – it should be a (local device/user) configurable thing as well.

In the case of MyList, I’ve found having all the sections collapsed but Yesterday, Today, and Tomorrow  - with the others collapsed is really (IMHO) a better way because they are dynamically created and serve as a reminder there are activities in other time slots (past and future) right at your fingertips.

## 

RJLedger – www.rjlsystems.nethttp://www.rjlsystems.net/
Blog: www.rjlsystems.net/blog.html
Direct Phone: +1 603.369.3047  (alt: +1  508.809.9513)

**Comment by rjledger** [comment]
@jbest84 - Just checked out the new "collapse" property/feature in DateTimeSection - just what the Dr. ordered - fixes my "ailment" ;-)

**Comment by rjledger** [comment]
Oops.. spoke a little too soon - was running a test against a customer's system we are doing a POC on for 8 and discovered if you have any "old" activities that are last year (or before) it does some strange things in the grouped list headers.
EX: 
  If the activity is last year and it is in Sept thru Dec they show up as "Unknown" 
  If the activity was last year (2012) in Aug - it shows up "blank"
  If the activity was the year before last year (2011) and it was in July it shows up as last month ... etc

My solution was to add a "Past Year(s)" grouping:

```
       pastYearText: 'Past Year(s)',

       this.sections.push({ key: 'PastYear', title: this.pastYearText, value: null, collapsed: true })

       //First one checked
       if (this._isPastYear(valueDate)) {
            return "PastYear";
        }
        //followed by "_isOlder"

    isPastYear: function(value) {
        return value.getFullYear() < this.currentYear;
    },
```

Now it groups nicely..  All old activities before the "current year" fall into the "Past Year(s)" grouping ;-)

**Comment by jbest84** [comment]
I did some more cleanup on this today. The "two" and "three" month and week categories are gone. "This year" is now "Earlier this year" and "Later this year" to match what I changed with week. I added a "Past years(s)" group as well. All of the methods re-factored to take advantage of the momentjs lib we added. They should no longer rely on the order that the if statement appears. Check out the changes here - https://github.com/Saleslogix/argos-sdk/commit/dd1f570fc1b3e1f813764a2f3c444795794f3bfd
